ykcompile/src/lib.rs

```diff
@@ -169,51 +169,56 @@ impl<TT> TraceCompiler<TT> {
 
     fn place_to_location(&mut self, p: &Place) -> Result<Location, CompileError> {
         if !p.projection.is_empty() {
-            if p.projection.len() > 1 {
-                todo!("chains of projections");
-            }
+            self.resolve_projection(p)
+        } else {
+            self.local_to_location(p.local)
+        }
+    }
 
-            // FIXME deal with remaining projections and chains of projections.
+    /// Takes a Place, resolves all projections, and returns a Location containting the result.
+    fn resolve_projection(&mut self, p: &Place) -> Result<Location, CompileError> {
+        let mut curloc = self.local_to_location(p.local)?;
+        if p.projection.len() > 1 {
+            todo!("Deal with remaining projections");
+        }
+        for proj in &p.projection {
```

Comment:
Can we deal with chains yet? Is this loop just in preparation for when we can?

Reply:
> This PR refactors the code that deals with projections and prepares us for being able to deal with chains of projections soon (we'll need types of projections for that though).

I initially planned for this PR to deal with chains of projections, but then decided to keep this a refactoring PR that doesn't add new functionality.

Reply:
ok